### DISCUSSION TITLE
Using PointCloud

### DISCUSSION THREAD

**Discussion by ruchitkini** [lululxvi/deepxde#1509] [Q&A] [answered]
Hi LuLu,

I am trying to use deepxde to model a complex 3D geometry using PointCloud method. For initial trials, I tried using a 2D geometry. I imported the mesh points from Gmsh, extracted the boundary points and inside points, and gave them as inputs of geometry. After training the model, the train loss and the test loss are the same. It seems like the model takes test points the same as the train points. Why is that so?

Below is my code to extract the node and boundary coordinates:
```
import numpy as np
import gmsh
import sys
import matplotlib.pyplot as plt

gmsh.initialize()

gmsh.open('2D_test2.msh')

dim = -1
tag = -1 
nodeTags, coords, parametricCoord = gmsh.model.mesh.getNodes(dim, tag)

coords = coords.reshape(-1, 3)
coords = np.delete(coords, 2, 1)

# get boundary coordinates
boundary_node_tags, boundary_node_coords = gmsh.model.mesh.getNodesForPhysicalGroup(-1, 13) 
coords_boundary = boundary_node_coords.reshape(-1, 3)
coords_boundary = np.delete(coords_boundary, 2, 1)

gmsh.finalize()
print(len(coords))
print(len(coords_boundary))

fig, ax = plt.subplots(figsize = (6, 3))
ax.axis('equal')
ax.scatter(coords[:, 0], coords[:, 1], s = 0.2)
plt.show()
```

Here is the code to train the model:
```
import deepxde as dde

def pde(x, y):
    # Most backends
    dy_xx = dde.grad.hessian(y, x, i=0, j=0)
    dy_yy = dde.grad.hessian(y, x, i=1, j=1)

    return -dy_xx - dy_yy - 1

def boundary(_, on_boundary):
    return on_boundary

geom = dde.geometry.PointCloud(coords, boundary_points = coords_boundary)

bc = dde.icbc.DirichletBC(geom, lambda x: 0, boundary)

data = dde.data.PDE(geom, pde, bc, num_domain=0, num_boundary=0, num_test=1000, anchors = coords)
net = dde.nn.FNN([2] + [50] * 4 + [1], "tanh", "Glorot uniform")
model = dde.Model(data, net)

model.compile("adam", lr=0.001)
model.train(iterations=3000)
model.compile("L-BFGS")
losshistory, train_state = model.train(display_every = 1000)
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```
I am using the 2D Poisson's equation example, for my geometry.

After plotting the train points and test points the test points look the same as the train points and overlap the previous points. Also, how are the test points assigned, with random distribution or any other distribution? I could not find anything in the documentation.

I also had another question: To get the new set of points within the geometry, how to achieve that? I found a code snippet in the documents which are shown below:
`x_new = geom.uniform_points(1000, boundary = True)`
Does this code only take the points from the set of original train points or new points inside the defined geometry?

**Comment by lululxvi** [comment, marked as answer]
For point cloud, all the training and testing points are sampled for the provided points.
